### PR TITLE
Port filter v13

### LINKED ISSUE #246: Port filter command to v13

### PR DESCRIPTION
**Pull request by Shaylin-8bit**
## What issue is this solving?

Closes #246 

### Description

Ported filter to work in V13 and reformatted.

Removed auto moderation functions since we've gone on the path of not using them. 
Rewrote to use `RoleEnum` instead of `isHighRoller`.
Rewrote conversion function to use an object, and `.map()` instead of the spaghetti mess of `||`s.

## Any helpful knowledge/context for the reviewer?

- Is a re-seeding of the database necessary? 
   no
- Any new dependencies to install? 
   Uses bad-words library from https://www.npmjs.com/package/bad-words
- Any special requirements to test? 
  Filter ignores any account with code-counselor or higher, so test account cannot have those.

### Please make sure you've attempted to meet the following coding standards
- [x] Code has been tested and does not produce errors
- [x] Code is readable and formatted
- [x] There isn't any unnecessary commented-out code

### CODE REVIEW

File: commands/moderation/filter.js [hunk 1]
```diff
@@ -1,122 +1,61 @@
...
-          logMsg(msg, 'word', check[0]);
-        }
-      }
+    if (RoleEnum[msg.member.roles.highest.name]) return;
+
+    const profanity = new Filter();
+    const spam = new Filter({emptyList: true});
+    spam.addWords(...['nitro']);
+    const content = convert(msg.content);
+    const profane = profanity.isProfane(content);
```
[the remark this answers is not in the document]
Comment (Shaylin-8bit) [reply commands/moderation/filter.js]: Excellent suggestion! Thanks and done!

File: commands/moderation/filter.js [hunk 2]
```diff
@@ -1,122 +1,61 @@
...
-      word = words[i];
-    }
+    logs.send({content: reply});
   }
-  return word ? [word, null] : [null, null];
 };
 
-const spamCheck = (content) => {
-  return content.includes('free') && content.includes('nitro');
+const convert = (sentence) => {
```
[the remark this answers is not in the document]
Comment (Shaylin-8bit) [reply commands/moderation/filter.js:39]: I went with using a `new Map()` and seems to have gotten past eslint/prettier, which ever was formatting it wrong.

File: commands/moderation/filter.js [hunk 3]
```diff
@@ -1,122 +1,48 @@
...
-
-  for (let i = 0; i < words.length; i++) {
-    if (strongWords.test(words[i])) {
-      return [words[i], 'strong'];
-    } else if (lightWords.test(words[i])) {
-      word = words[i];
-    }
-  }
-  return word ? [word, null] : [null, null];
-};
```
Comment (aedwardg) [inline commands/moderation/filter.js:118]: I ❤️ deleted code!

File: config/filters.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const Filter = require('bad-words');
+const profanity = new Filter();
+const spam = new Filter({emptyList: true});
+spam.addWords(...['nitro']);
+
+const map = new Map([
+  ['@', 'a'],
+  ['4', 'a'],
```
Comment (aedwardg) [inline config/filters.js:8]: This refactor is a huge improvement, great job.

**Review by aedwardg — APPROVED** [review]
Looks great @Shaylin-8bit!

File: commands/moderation/filter.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mikejoh12) [reply commands/moderation/filter.js:38]: Great!

**Review by mikejoh12 — APPROVED** [review]
Great work!

File: config/filters.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  ['0', 'o'],
+  ['7', 't'],
+  ['3', 'e']
+]);
+
+module.exports = {
+  getProfanity: () => profanity,
+  getSpam: () => spam,
+  getMap: () => map,
+};
```
Comment (Victoria-DR) [inline config/filters.js:24]: Missing a newline at end of file.

**Review by Victoria-DR — APPROVED** [review]
Amazing! Left one small comment for a missing newline at the end of `filters.js`, but approved in the meantime.

File: commands/moderation/filter.js (hunk 3 above)
Comment (Shaylin-8bit) [reply commands/moderation/filter.js:118]: Same 💯

**Pull request merged**